lib/network_ssl/network_ssl_compat.c

```diff
@@ -0,0 +1,208 @@
+#include <assert.h>
+#include <limits.h>
+#include <stddef.h>
+#include <string.h>
+
+#include <openssl/ssl.h>
+
+#include "network_ssl_compat.h"
+
+#ifdef NETWORK_SSL_COMPAT_TLS_VERSION
+/**
+ * network_ssl_compat_TLS_client_method():
+ * Create a SSL_METHOD for TLS 1.2.
+ *
+ * COMPATIBILITY: provides TLS_client_method(), but only for TLS 1.2 (not
+ * based on negotiation with the server).
+ */
+const SSL_METHOD *
+network_ssl_compat_TLS_client_method()
+{
+
+	return TLSv1_2_client_method();
```

Comment:
According to `STYLE`, I guess this should be
```
return (TLSv1_2_client_method());
```
[because you give the example of `return (baz(bar));`]

Is that really what you want?

Reply:
It's a bit ugly, but yeah let's be consistent.

Reply:
Shouldn't we be using the "version-flexible" SSLv23_client_method here though?

Reply:
As long as we're totally certain that `_set_min_proto_version` will fail if there's any problem disabling the protocols which are vulnerable to poodles and beasts, I guess so.